pubtools/exodus/hooks/pulp.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import os
+import sys
+
+import backoff
+import requests
+from pubtools.pluggy import hookimpl, pm  # pylint: disable=wrong-import-order
+from six.moves.urllib.parse import urljoin
+
+LOG = logging.getLogger("pubtools-exodus")
+
+
+class ExodusPulpHanlder:
+    def __init__(self):
+        self._gw_url = os.getenv("EXODUS_GW_URL")
+        self._gw_env = os.getenv("EXODUS_GW_ENV")
+        self._gw_crt = os.getenv("EXODUS_GW_CERT")
+        self._gw_key = os.getenv("EXODUS_GW_KEY")
+
+        self._publish = None
+        self._check_cert()
```

Comment:
I think anything which does an HTTP request or other slow or possibly failing steps needs to be delayed until it's time to create a publish.

Keep in mind that task_start hook is invoked for all kinds of tasks, including those which have nothing to do with Pulp or Exodus at all.  For example pubtools-ami, pubtools-quay tasks will end up hitting this code too.  So you don't want to waste time, produce confusing log messages or introduce possibility of failure until you know for sure that this task has something to do with exodus & pulp.